scalameta/trees/shared/src/main/scala/scala/meta/internal/prettyprinters/TreeSyntax.scala

```diff
@@ -296,6 +296,65 @@ object TreeSyntax {
       }
       !isOnlyChildOfOnlyChild(t)
     }
+    def guessNeedsLineSep(t: Template): Boolean =
+      t.stats.isEmpty
+    @annotation.tailrec
+    def guessNeedsLineSep(t: Stat): Boolean = t match {
+      case _: Decl.Val => true
+      case _: Decl.Var => true
+      case _: Decl.Def => true
+      case _: Decl.Type => false
+      case Defn.Val(_, _, _, rhs) => guessNeedsLineSep(rhs)
+      case Defn.Var.After_4_7_2(_, _, _, body) => guessNeedsLineSep(body)
+      case Defn.Def.After_4_7_3(_, _, _, _, body) => guessNeedsLineSep(body)
```

Comment:
I used the versioned `unapply` because I was using Intellij at first and the default apply has showing as deprecated, that was the only reason. I'll use your sugestions to improve the code, thank you!

Reply:
understood. yes, for some classes, `unapply` was preserved for MIMA compatibility and made deprecated. but let's not use `unapply` :)